### PR TITLE
ci: Move MacOS builds to CircleCI (5-0-x)

### PR DESCRIPTION
**Pull request by jkleinsc**
#### Description of Change
This PR puts back the changes to move MacOS builds to CircleCI for 5-0-x.  They were reverted due to issues found during release.  This also pulls in #16766 and #16824 which should resolve the release issues.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: no-notes

### CODE REVIEW

**Comment by jkleinsc** [comment]
Release test builds here:
CircleCI release build request for osx-publish successful.  Check https://circleci.com/gh/electron/electron/132582 for status.
CircleCI release build request for mas-publish successful.  Check https://circleci.com/gh/electron/electron/132583 for status.

File: script/ci-release-build.js [hunk 1]
```diff
@@ -14,7 +14,9 @@ const circleCIJobs = [
   'linux-arm-publish',
   'linux-arm64-publish',
   'linux-ia32-publish',
-  'linux-x64-publish'
+  'linux-x64-publish',
+  'mas-publish',
+  'osx-publish'
 ]
 
 const vstsJobs = [
```
Comment (MarshallOfSound) [inline script/ci-release-build.js]: There's another commit somewhere that removes the VSTS jobs from this file, we need that commit as well

Comment (MarshallOfSound) [reply script/ci-release-build.js]: https://github.com/electron/electron/commit/a569dad8c52303e292597c9005ec470eb97d13bc

Comment (jkleinsc) [reply script/ci-release-build.js]: Good catch @MarshallOfSound.  Cherry pick on its way.

**Comment by MarshallOfSound** [comment]
We are currently running a 6.0.0-nightly off of master, if that succeeds we can merge this 👍

**Comment by codebytere** [comment]
@jkleinsc can you add https://github.com/electron/electron/pull/16893 to this?

**Pull request merged**